### DISCUSSION TITLE
What is impact on the image embedding result when the alpha channel is dropped by the converter

### DISCUSSION THREAD

**Discussion by githubusersel** [mlfoundations/open_clip#800] [Q&A]
Hi everyone,

- I'm wondering what is the impact on image embedding result when alpha channel dropped by converter? I found that taking screenshots from Mac adds extra alpha channel to the image. Because CLIP model only works with RGB, it needs to be converted. But I don't think there's enough explanation of how it affects the embedding results using the dropped channel. 
- What's good practice to convert RGBA to RGB? It looks like convert method from Image module was used in ReadMe. Is there more efficient way?
` im = Image.open("cat.jpg").convert("RGB")`

**Comment by rwightman** [comment]
@githubusersel the 'correct' way is to create an image with the desired background color and then paste the RGB portion of the image onto that background image using the A channel as a mask...

See https://stackoverflow.com/questions/9166400/convert-rgba-png-to-rgb-with-pil

Unfortunately, the correct background color is image specific. Although I'd bet white would be more correct on avg than black for an RGBA, there would certainly be some RGBA images out there intended for dark backgrounds...
